packages/mongodb-artifact-generator/src/chat/makeGenerateResponse.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  ChatRequestMessage,
+  FunctionDefinition,
+  OpenAIClient,
+} from "mongodb-rag-core";
+import { FormattedJiraIssue } from "../commands/generateJiraPromptResponse";
+import { RunLogger } from "../runlogger";
+import {
+  asJsonSchema,
+  formatFewShotExamples,
+  formatMessagesForArtifact,
+  PromptExamplePair,
+} from "./utils";
+import { z } from "zod";
+import { stripIndents } from "common-tags";
+import { Summary } from "./makeSummarizer";
+
+export type GeneratedResponse = z.infer<typeof GeneratedResponse>;
+export const GeneratedResponse = z.object({
+  response: z.string().describe("The generated response text."),
+});
+
+const generateResponseTool: FunctionDefinition = {
+  name: "generateResponse",
+  description: "A response generated based on a given context.",
+  parameters: asJsonSchema(GeneratedResponse),
+};
+
+export type MakeGenerateResponseArgs = {
+  openAi: {
+    client: OpenAIClient;
+    deployment: string;
+  };
+  logger?: RunLogger;
+  directions?: string;
+  examples?: PromptExamplePair[];
+};
+
+export type GenerateResponseArgs = {
+  issue: FormattedJiraIssue;
+  summary: Summary;
+  prompt: string;
+};
+
+export function makeGenerateResponse({
+  openAi,
+  logger,
+  directions,
+  examples = [],
+}: MakeGenerateResponseArgs) {
+  return async function generateResponse({
+    issue,
+    summary,
+    prompt,
+  }: GenerateResponseArgs) {
+    const messages = [
+      {
+        role: "system",
+        content: [
+          `Your task is to generate a response to a provided input. The response should be relevant to the input and based only on the provided context.`,
+          directions ?? "",
+        ].join("\n"),
+      },
+      ...formatFewShotExamples({
+        examples,
+        functionName: generateResponseTool.name,
+        responseSchema: GeneratedResponse,
+      }),
+      {
+        role: "user",
+        content: JSON.stringify({ issue, summary, prompt }),
+      },
+    ] satisfies ChatRequestMessage[];
+    const result = await openAi.client.getChatCompletions(
```

Comment:
using old oai client it seems. i rec switching to new. but maybe we should do that in RAG core and propagate everywhere..think we have separate ticket for that

Reply:
Yeah that's my thought too - probably easier to do as an independent unit where we upgrade everywhere